gcloud/test_connection.py

```diff
@@ -66,6 +67,26 @@ def test_user_agent_format(self):
         conn = self._makeOne()
         self.assertEqual(conn.USER_AGENT, expected_ua)
 
+    def test_thread_local_http(self):
+        credentials = _Credentials(lambda x: object())
+        conn = self._makeOne(credentials)
+
+        self.assertTrue(conn.http is not None)
+
+        # Should return the same instance when called again.
+        self.assertTrue(conn.http is conn.http)
+
+        # Should return a different instance from a different thread.
+        http = conn.http
+
+        def test_thread():
+            self.assertTrue(conn.http is not None)
+            self.assertTrue(conn.http is not http)
+
+        thread = threading.Thread(target=test_thread)
+        thread.start()
```

Comment:
Indeed. I [fooled around][1] a bit to "prove" to myself I knew what was happening.

IMO two threads still a good idea.

(We'd probably also want to add some kind of multi-threaded torture test, though I'm not sure how often/where we'd run it.)

[1]: https://gist.github.com/dhermes/60d3294c31cb359a743a